Documentation/Integration.md

```diff
@@ -2,29 +2,111 @@
-* _TODO: complete this documentation_
+
+Transfer Extensions are located in the [extensions/data-transfer module](https://github.com/google/data-transfer-project/tree/master/extensions/data-transfer), where they are organized by service provider and then by data type.
+
+1. Create a new directory for your provider under extensions/data-transfer
+
+  `$mkdir extensions/data-transfer/portability-data-transfer-foo`
+
+2. Create a build.gradle file in your new directory
+
+  `$touch extensions/data-transfer/portability-data-transfer-foo/build.gradle`
+
+3. Add the spi and cloud dependencies to your build file along with any sdk or related dependencies specific to your code
+
+  * Common dependencies include logging, see example below
+
+  ```javascript
+  /*
+   *  Copyright 2018 The Data Transfer Project Authors.
+   * 
+   *  Licensed under the Apache License, Version 2.0 (the "License");
+   *  you may not use this file except in compliance with the License.
+   * You may obtain a copy of the License at
+   *
+   * https://www.apache.org/licenses/LICENSE-2.0
+   *
+   * Unless required by applicable law or agreed to in writing, software
+   * distributed under the License is distributed on an "AS IS" BASIS,
+   * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   * See the License for the specific language governing permissions and
+   * limitations under the License.
+   */
+
+  dependencies {
+      compile project(':portability-spi-cloud')
+      compile project(':portability-spi-transfer')
+      
+     // logging
+      compile("org.slf4j:slf4j-api:${slf4jVersion}")
+      compile("org.slf4j:slf4j-log4j12:${slf4jVersion}")
+     
+      compile(YOUR DEPS HERE)
+      }
+  ```
+  
+4. Add an entry for your extension in root settings.gradle
+``` javascript
+// Foo transfer extension
+include ':extensions:data-transfer:portability-data-transfer-foo'
+```
+
+5. Create the META-INF file
+
+  `$touch extensions/data-transfer/portability-data-transfer-foo/src/main/resources/META-INF/services/org.dataportabilityproject.spi.transfer.extension.TransferExtension`
+
+6. Create the directory for your main code
+
+ `$mkdir -p extensions/data-transfer/portability-data-transfer-foo/src/main/java/org/dataportabilityproject/transfer/foo/`
+
+7. Create FooTransferExtension extending TransferExtension
+
+  * See class javadoc for TransferExtension for full documentation
+
+  * Best practices:
+
+    * Validate the TransferExtension isn’t validated more than once in a running binary
+    * Most providers will interact with their api via a ‘registered’ app providing a client id/key and secret. Those will be stored in a key management system and retrieved via an identifier for the key and one for the secret,
+    * In initialize, retrieve the AppCredentialStore
+    * Return an AuthDataGenerator for a given transfer type and mode
```

Comment:
returning an authdatagenerator is done in the AuthExtension below. 
This should be return an importer/exporter based on the data type.

Reply:
Done. PTAL